hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/rejecttokens/RejectTokensDecoderTest.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    @Test
+    void decodeHtsCallNFT() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(LedgerConfig.class)).willReturn(ledgerConfig);
+        given(ledgerConfig.tokenRejectsMaxLen()).willReturn(10);
+        given(addressIdConverter.convert(asHeadlongAddress(SENDER_ID.accountNum())))
+                .willReturn(SENDER_ID);
+
+        final var encoded = Bytes.wrapByteBuffer(RejectTokensTranslator.TOKEN_REJECT.encodeCall(
+                Tuple.of(asHeadlongAddress(SENDER_ID.accountNum()), new Address[] {}, new Tuple[] {
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L)
+                })));
+
+        // when
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        final var expected = TokenRejectTransactionBody.newBuilder()
+                .owner(SENDER_ID)
+                .rejections(TokenReference.newBuilder()
+                        .nft(NftID.newBuilder()
+                                .tokenId(FUNGIBLE_TOKEN_ID)
+                                .serialNumber(1L)
+                                .build())
+                        .build())
+                .owner(SENDER_ID)
+                .build();
+
+        // then
+        final var decoded = subject.decodeTokenRejects(attempt);
+
+        assertNotNull(decoded);
+        assertEquals(expected, decoded.tokenReject());
+    }
+
+    @Test
+    void decodeFailsIfReferencesExceedLimits() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(LedgerConfig.class)).willReturn(ledgerConfig);
+        given(ledgerConfig.tokenRejectsMaxLen()).willReturn(10);
+
+        final var encoded = Bytes.wrapByteBuffer(RejectTokensTranslator.TOKEN_REJECT.encodeCall(Tuple.of(
+                OWNER_HEADLONG_ADDRESS,
+                new Address[] {
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS
+                },
+                new Tuple[] {
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L),
+                })));
+
+        // when
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        // then
+        assertThrows(HandleException.class, () -> subject.decodeTokenRejects(attempt));
```

Comment:
Consider the more complete asserts as for the token cancel PR

Reply:
Done